#25 Rons templates branch

FrugalGuy opened the pull request:
Issue #25 asks for PR how-to. 
add CONTRIBUTING.md
add CODE_OF_CONDUCT.md
add doc/contributor_license_agreement.md

FrugalGuy committed:
docs:add CONTRIBUTING.md

Add instructions on how to contribute fixes and features to the project.

Fixes #25

FrugalGuy committed:
docs:add CODE_OF_CONDUCT.md update CONTRIBUTING.md

Add code of conduct and update CONTRIBUTING to point to it.

Fixes #25

FrugalGuy committed:
issue25: update pull request directions

FrugalGuy committed:
issue35: add Contrib Lic Agreement to doc/

FrugalGuy committed:
issue35: update Contrib Lic Agreement

FrugalGuy commented:
**CONTRIBUTING.md**

1. I think we trash the whole Commit Message format section. We're not doing automated release notes from commit messages. 
2. Update Signing the CLA to note how that is done (during PR creation).

**doc/contributor_license_agreement.md**
1. Delete the file since it's now stored outside the repo where the web hook can get to it
2. Add a link to the agreement so folks can read it before doing PRs, if they want.

FrugalGuy commented:
PR directions start at number 6.  I need to re-work that to be structured more clearly on the page.

petesoper commented:
Some initial remarks. More to come.
1. Quarcus instead of Quercus in contributor_license_agreement.md
2. "In order to reproduce bugs, we require that you provide a minimal reproduction." How about "if at all possible", 'cause sometimes reproduction is not possible. For instance when a proprietary system is involved the submitter may not be able to legally make available the source files or hardware specs required for reproduction. 
3. "Unfortunately, we are not able to investigate / fix bugs without a minimal reproduction" How about "may not be able to".  If somebody sends a bug report that sez "Line 24 of file foo.c is broken: needs to be +=, not =" there is no bug reproduction involved. There are other cases where a problem is described but the submitter cannot (legally, practically, etc) provide everything required to reproduce the problem: they can only describe their environment to a certain level of detail and its on us to try to reproduce it, typically with a dialog with the user. Yes, if the user submits the bug, disappears and does not answer questions and we hit a dead end, that’s the end of it, I agree.
4. Quarcus again in the contributor agreement. This agreement is mighty heavy! I don't understand the copyright section as it relates to how the copyright lines (one each for hardware license and software license) get changed as a result of an individual's contribution.
5. I agree the commit message format section should be slashed down. How about a few items:
   1. Include #N where N is the issue the commit relates to and
   2. Descriptive text summarizing the changes
   3. “doc:” for doc changes and perhaps a few other labels like this

FrugalGuy committed:
#35 Delete cla from doc folder

CLA is stored externally at https://gist.github.com/FrugalGuy/2ffa493e842ce030d9384caea7d00971
CLA spelling errors corrected.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/TriEmbed/quercus?pullRequest=39) <br/>All committers have signed the CLA.
